include/opengemini/impl/enc/LineProtocolEncoder.cpp

```diff
@@ -0,0 +1,185 @@
+//
+// Copyright 2024 Huawei Cloud Computing Technologies Co., Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#include "opengemini/impl/enc/LineProtocolEncoder.hpp"
+
+#include <type_traits>
+
+#include "opengemini/Exception.hpp"
+#include "opengemini/Precision.hpp"
+
+namespace opengemini::impl::enc {
+
+namespace {
+
+template<typename DURATION>
+inline auto Round(const Point::Time& time)
+{
+    return std::chrono::time_point_cast<Point::Time::duration>(
+               std::chrono::round<DURATION>(time))
+        .time_since_epoch()
+        .count();
+}
+
+} // namespace
+
+OPENGEMINI_INLINE_SPECIFIER
+LineProtocolEncoder::LineProtocolEncoder()
+{
+    os_.precision(12);
+}
+
+OPENGEMINI_INLINE_SPECIFIER
+std::string LineProtocolEncoder::Encode(const Point& point)
+{
+    AppendPoint(point);
+    return os_.str();
+}
+
+OPENGEMINI_INLINE_SPECIFIER
+std::string LineProtocolEncoder::Encode(const std::vector<Point>& points)
+{
+    if (points.empty()) {
+        throw Exception(errc::LogicErrors::InvalidArgument,
```

Comment:
This should be sync with other languages

Reply:
Now we allow user to pass an empty vector of points that the library will neither throw an exception nor invoke any request to server for it.